src/bufr/DataObject.cpp

```diff
@@ -0,0 +1,48 @@
+/*
+ * (C) Copyright 2021 NOAA/NWS/NCEP/EMC
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+#include "DataObject.h"
+
+#include <iostream>
+
+#include "eckit/exception/Exceptions.h"
+
+
+namespace Ingester
+{
+    std::shared_ptr<DataObjectBase>
+        DataObjectBase::fromResult(const std::shared_ptr<bufr::ResultBase>& resultBase,
+                                   const std::string& query)
+    {
+        std::shared_ptr<DataObjectBase> dataObject = nullptr;
+
+        if (auto result = std::dynamic_pointer_cast<bufr::Result<float>>(resultBase))
```

Comment:
Only float and str? What about int?

Reply:
**query_cxx** only supports float and string. The other branch, **query_cxx_types** (merging later) actually refactors a lot of this (this function disappears) and has the flexibility to support basically any type.